test/fileio.jl

```diff
@@ -50,6 +50,30 @@ let fn = tempname() * ".h5"
   dat = load(fn; dict=OrderedDict())  # `track_order` is inferred from `OrderedDict`
 
   @test all(keys(dat) .== ["b", "a", "G/z", "G/f"])
+
+  # issue #939
+  h5open(fn, "r"; track_order=true) do io
+    @test HDF5.get_context_property(:file_create).track_order
+    @test all(keys(io) .== ["b", "a", "G"])
+    @test HDF5.get_context_property(:group_create).track_order
+    @test HDF5.get_create_properties(io["G"]).track_order  # inferred from file, created with `track_order=true`
+    @test all(keys(io["G"]) .== ["z", "f"])
+  end
+
+  h5open(fn, "r"; track_order=false) do io
+    @test !HDF5.get_context_property(:file_create).track_order
+    @test all(keys(io) .== ["G", "a", "b"])
+    @test !HDF5.get_context_property(:group_create).track_order
+    @test HDF5.get_create_properties(io["G"]).track_order  # inferred from file
+    @test all(keys(io["G"]) .== ["z", "f"])
+  end
+
+  h5open(fn, "r") do io
+    @test !HDF5.get_create_properties(io).track_order
```

Comment:
If track order is false can't the check on line 73 fail since the keys are unordered? Wouldn't an equality check using `Set` be safer?

Reply:
In this case, `track_order` from `file_create` is inferred from file so this test is correct.